browser/notifications/brave_notification_platform_bridge_mac.mm

```diff
@@ -0,0 +1,91 @@
+#include "base/strings/utf_string_conversions.h"
+#include "base/task/post_task.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/notifications/notification_common.h"
+#include "chrome/browser/notifications/notification_display_service_impl.h"
+#include "chrome/browser/notifications/platform_notification_service_impl.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/profiles/profile_manager.h"
+#include "chrome/browser/ui/cocoa/notifications/notification_builder_mac.h"
+#include "chrome/browser/ui/cocoa/notifications/notification_constants_mac.h"
+#import "chrome/browser/ui/cocoa/notifications/notification_delivery.h"
+#import "chrome/browser/ui/cocoa/notifications/notification_response_builder_mac.h"
+#include "chrome/common/buildflags.h"
+#include "chrome/grit/generated_resources.h"
+#include "components/crash/content/app/crashpad.h"
+#include "components/url_formatter/elide_url.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
+#include "third_party/blink/public/platform/modules/notifications/web_notification_constants.h"
+#include "third_party/crashpad/crashpad/client/crashpad_client.h"
+#include "ui/base/l10n/l10n_util_mac.h"
+#include "ui/message_center/public/cpp/notification.h"
+#include "ui/message_center/public/cpp/notification_types.h"
+#include "url/gurl.h"
+#include "url/origin.h"
+
+@class NSUserNotificationCenter;
+
+// static
+std::unique_ptr<NotificationPlatformBridge>
+NotificationPlatformBridge::Create() {
+  base::scoped_nsobject<BraveAlertDispatcher> alert_dispatcher(
+      [[BraveAlertDispatcher alloc] init]);
+  return std::make_unique<BraveNotificationPlatformBridgeMac>(
+      [NSUserNotificationCenter defaultUserNotificationCenter],
+      alert_dispatcher.get());
+}
+
+BraveNotificationPlatformBridgeMac::BraveNotificationPlatformBridgeMac(
+    NSUserNotificationCenter* notification_center,
+    id<AlertDispatcher> alert_dispatcher)
+    : NotificationPlatformBridgeMac(notification_center, alert_dispatcher) {
+  notification_platform_bridge_notification_timeout_ =
+      [[[NotificationPlatformBridgeNotificationTimeoutMac alloc] init] retain];
+}
+
+BraveNotificationPlatformBridgeMac::~BraveNotificationPlatformBridgeMac() {
+  [notification_platform_bridge_notification_timeout_ release];
+  notification_platform_bridge_notification_timeout_ = nil;
+}
+
+void BraveNotificationPlatformBridgeMac::Display(
+    NotificationHandler::Type notification_type,
+    Profile* profile,
+    const message_center::Notification& notification,
+    std::unique_ptr<NotificationCommon::Metadata> metadata) {
+  NotificationPlatformBridgeMac::Display(
+      notification_type, profile, notification, metadata);
+
+  if (!IsPersistentNotification(notification) && notification_type != NotificationHandler::Type::BRAVE_ADS) {
```

Comment:
you'll need to call `NSUserNotification* toast = [builder buildUserNotification];` again before this

Reply:
Fixed